tests/unit/r2i/tflite/parameters.cc

```diff
@@ -0,0 +1,263 @@
+/* Copyright (C) 2018-2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#include <r2i/r2i.h>
+#include <r2i/tflite/parameters.h>
+#include <r2i/tflite/engine.h>
+#include <r2i/tflite/prediction.h>
+
+#include <CppUTest/CommandLineTestRunner.h>
+#include <CppUTest/TestHarness.h>
+
+namespace mock {
+
+class Engine : public r2i::IEngine {
+ public:
+  r2i::RuntimeError Start () {
+    return r2i::RuntimeError();
+  };
+  r2i::RuntimeError Stop () {
+    return r2i::RuntimeError();
+  };
+  r2i::RuntimeError SetModel (std::shared_ptr<r2i::IModel> in_model) {
+    return r2i::RuntimeError();
+  };
+  virtual std::shared_ptr<r2i::IPrediction> Predict (std::shared_ptr<r2i::IFrame>
+      in_frame, r2i::RuntimeError &error) { return nullptr; }
+};
+}
+
+namespace r2i {
+namespace tflite {
+
+Engine::Engine ()  { }
+RuntimeError Engine::SetModel (std::shared_ptr<IModel> in_model) { return RuntimeError(); }
+RuntimeError Engine::Start () { return RuntimeError(); }
+RuntimeError Engine::Stop () { return RuntimeError(); }
+
+RuntimeError Engine::SetNumberOfThreads (int name) {
+  this->number_of_threads = name;
+  return RuntimeError();
+}
+const int Engine::GetNumberOfThreads () { return this->number_of_threads; }
+
+RuntimeError Engine::SetAllowFP16 (int name) {
+  this->allow_fp16 = name;
+  return RuntimeError();
+}
+const int Engine::GetAllowFP16 () { return this->allow_fp16; }
+
+std::shared_ptr<IPrediction> Engine::Predict (std::shared_ptr<IFrame> in_frame,
+    RuntimeError &error) { return nullptr; }
+Engine::~Engine () { }
+
+}
+}
+
+TEST_GROUP (TliteParameters) {
+};
+
+TEST (TliteParameters, ConfigureIncompatibleEngine) {
+  r2i::RuntimeError error;
+  r2i::tflite::Parameters parameters;
+
+  std::shared_ptr<r2i::IEngine> engine(new mock::Engine);
```

Comment:
you are using these almost in every test, I recommend factoring them out to the setup and teardown

Reply:
This is not resolved